Integration test ci

kleash opened the pull request:
- Integration test for installing wasmcc and running sample rust contract
- Updated github actions ci

Fixes issue #4

kleash committed:
integration tests
- Use nwo for bootstrapping fabric network
- Use solo network setup
- install wasmcc
- install sample rust wasm chaincode
- invoke and query installed chaincode

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>

MHBauer commented:
@swetharepakula this looks like it's running. It takes 2 minutes. WDYT?

MHBauer reviewed (approved):
Given the CI looks like it's running, I'm go, but I'd wait for one more review from @swetharepakula if you can hold for a bit.

MHBauer commented on .github/workflows/go.yml:
```diff
@@ -14,7 +14,55 @@ jobs:
+    - name: Integration test
+      working-directory: ./integration/e2e
+      run: |
+        pwd
```
maybe not this

kleash replied:
removed pwd

MHBauer commented:
Only thing I'd prefer is to get the local run of this the same as CI. I find it difficult to run locally.

swetharepakula commented on .github/workflows/go.yml:
```diff
@@ -14,7 +14,55 @@ jobs:
+        env
+        echo "###### GO ENV #######"
+        go env
+        echo ${GOPATH}
```
Is this piece necessary for ci?

kleash replied:
Nope, just added it for more logs. I will remove it

swetharepakula commented on .github/workflows/go.yml:
```diff
@@ -14,7 +14,55 @@ jobs:
+      working-directory: ./integration/e2e
+      run: |
+        pwd
+        go test
```
do you want to use ginkgo here instead since you install it above?

kleash replied:
go bin is not in path, that's why I avoided it. Is there any advantage of using ginkgo over go test?

kleash replied:
plus it needs gcc: https://github.com/kleash/fabric-chaincode-wasm/runs/268514684

swetharepakula replied:
No real advantage, just curious as to why you were installing it above.

kleash replied:
Oh ok. I think I tried using ginkgo first and later replaced it with go test. But didn't remove the installation part.

I have removed ginkgo installation now. You are right, it's an extraneous step.

swetharepakula commented on .github/workflows/go.yml:
```diff
@@ -14,7 +14,55 @@ jobs:
+        mkdir hyperledger
+        cd hyperledger
+        mkdir fabric
+        cd fabric
```
You don't need to create the fabric directory

kleash replied:
correct, thanks for pointing out

swetharepakula commented:
So I think you how it is right now is fine, but I agree with Morgan. Perhaps it is useful to make a script for some of the intermediary steps so that this can be recreated locally as well.

kleash commented:
> So I think you how it is right now is fine, but I agree with Morgan. Perhaps it is useful to make a script for some of the intermediary steps so that this can be recreated locally as well.

Ok. The only main thing required to run this is Fabric docker images and HLF 1.4.0 in go path. Maybe I can add both these in a script?

kleash committed:
Updated git actions pipeline for integration test

- set gopath to git workspace
- checkout code to gopath src
- set gopath/bin to paths for ginkgo
- Checkout HLF to gopath
- pull ccenv,couchdb,kafka,zookeeper images for fabric nwo
- run integration test

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>

swetharepakula commented on integration/e2e/utils.go:
```diff
@@ -0,0 +1,85 @@
+	ChannelID string
+	Name      string
+	Ctor      string
+}
```
You don't need this struct and its corresponding methods. We needed it so we could get hex encoded output. So you can remove that piece.

kleash replied:
fixed

swetharepakula commented on integration/e2e/utils.go:
```diff
@@ -0,0 +1,85 @@
+	components.Build()
+	return components
+}
+
```
I added this function because we needed to build fab3. Since you don't, you could directly just call components.Build(). You don't need this extra helper function.

kleash replied:
removed it

swetharepakula commented on integration/e2e/e2e_test.go:
```diff
@@ -0,0 +1,157 @@
+	})
+})
+
+func RunQuerInstalledChaincodesExists(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channel string) {
```
missing the `y` in Query

kleash replied:
fixed. Thanks for pointing out

swetharepakula commented on integration/e2e/e2e_test.go:
```diff
@@ -0,0 +1,157 @@
+
+}
+
+func RunQuerInstalledChaincodesEmpty(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, channel string) {
```
Missing the `y` in Query

kleash replied:
fixed. Thanks for pointing out

swetharepakula commented on integration/e2e/e2e_test.go:
```diff
@@ -0,0 +1,157 @@
+
+const LongEventualTimeout = time.Minute
+
+var _ = Describe("E2e", func() {
```
E2E or e2e

kleash replied:
fixed

swetharepakula reviewed (approved):
There are a few nits & typos. I think there are some additional functions in your utils.go that you don't need. I don't mind merging as is and having follow up prs to fix the smaller things.

swetharepakula commented:
@kleash Rebase, and update the branch. Then we can go ahead and merge this.

kleash committed:
goimports fixes

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>

kleash committed:
go format checker
- script for running go imports for all directories except vendor
- install goimports in git actions
- run the script from git actions

Signed-off-by: Shubham Aggarwal <shubham6300@gmail.com>
